simbad/command_line/__init__.py

```diff
@@ -321,16 +322,35 @@ def _simbad_contaminant_search(args):
         ed.process_miller_arrays()
         ed.output_mtz(temp_mtz)
 
```

Comment:
I think below could be re-written much nice. Instead of the `if`-`elif`-`else` conditions, maybe do the following:

1. In `simbad/rotsearch/__init__.py` include this function
```python
def rotation_search_factory(method):
    method_lc = method.lower()
    if method_ld == "amore":
        from simbad.rotsearch import AmoreRotationSearch
        return AmoreRotationSearch
    elif method_ld == "phaser":
        from simbad.rotsearch import PhaserRotationSearch
        return PhaserRotationSearch
    else:
        raise ValueError("Unrecognised program entered to perform rotation search: %s", args.rot_program)
```

2. In `simbad/command_line/__init__.py` replace current code with:

```python
from simbad.rotsearch import rotation_search_factory
rotsearch_obj = rotation_search_factory(args.rot_program)
rotation_search = rotsearch_obj(<args>, <kwargs>)
rotation_search.run(
    args.cont_db, 
    nproc=args.nproc, 
    min_solvent_content=args.min_solvent_content,
    submit_qtype=args.submit_qtype,
    submit_queue=args.submit_queue,
    chunk_size=args.chunk_size,
    shres=args.shres,
    pklim=args.pklim, 
    npic=args.npic,
    rotastep=args.rotastep,
)
```
To avoid any errors with args and kwargs not present in either function, just add the **kwargs to the `run` function of both, meaning you can provide dead arguments that are not considered [**make sure to mention this in the docstring**]

Reply:
I'll implement this now, it looks like a nice way of doing it